src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Legacy/CSharpCodeParser.cs

```diff
@@ -30,6 +30,14 @@ internal class CSharpCodeParser : TokenizerBackedParser<CSharpTokenizer>
             builder.Description = Resources.AddTagHelperDirective_Description;
         });
 
+    internal static readonly DirectiveDescriptor UsingDirectiveDescriptor = DirectiveDescriptor.CreateDirective(
```

Comment:
Why do we need this? It's not referenced anywhere from the compiler unlike the other descriptors.

Reply:
We could move it into tooling code. I added it there only for consistency since all other ones are defined there as well. Please let me know if you have a strong preference to move it into tooling code. If not, it seems nice (even though unnecessary) to keep them together.

Reply:
IMO, which I understand holds no weight, we should leave it here. Directives should always be supplied by the compiler, and the fact that the compiler doesn't have a "proper" using directive is something that @chsienki mentioned, and I would personally liked to see fixed: See [teams conversation](https://teams.microsoft.com/l/message/19:22e5c65cf5c843bf8dc9f7ac2fe6da33@thread.skype/1707781275653?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=4ba7372f-2799-4677-89f0-7a1aaea3706c&parentMessageId=1707780715591&teamName=.NET%20Developer%20Experience&channelName=Razor&createdTime=1707781275653) so hopefully it will be used in the future at some point?

Reply:
It seems likely to me that we're going to end up redoing all of how `using`s are parsed. I'm ok with putting this in the compiler for now, but it may well end up going away.

Reply:
Sounds good, I was just curious about the reasons, thanks for the explanation.

Reply:
@333fred - I'm happy to move and adjust this in the future as needed. If that's your only concern, would you please sign off for compilers as well?